### ISSUE TITLE
Possible bug: a log of error logs

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [Genymobile/scrcpy#303]
I get this error logs very often whenever I use scrcpy:

`E/HWComposer: getLayerReleaseFence failed for display -1: Invalid display`

Can it be a bug of some sort?
Can you somehow avoid creating this error? It's almost "spamming" Logcat...

**Comment by rom1v** [comment]
I don't think it's related to _scrcpy_.

**Comment by AndroidDeveloperLB** [comment]
@rom1v It appears only if I use it.
When I close its window, I don't see any new logs of this sort.
Here's a video showing this issue (I do the same thing when the window is closed) :

[2018-10-21_19-18-24.zip](https://github.com/Genymobile/scrcpy/files/2499388/2018-10-21_19-18-24.zip)

Please try to do the same.

**Comment by rom1v** [comment]
> It appears only if I use it.

I'm not so surprised, [HWComposer](https://source.android.com/devices/graphics/implement-hwc) is related to hardware compositing. I don't have this error in my logs.

**Comment by ENANHELL1** [comment]
21 окт. 2018 г. 19:28 пользователь "Romain Vimont" <notifications@github.com>
написал:

It appears only if I use it.

I'm not so surprised, HWComposer
<https://source.android.com/devices/graphics/implement-hwc> is related to
hardware compositing. I don't have this error in my logs.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Genymobile/scrcpy/issues/303#issuecomment-431682880>,
or mute the thread

**Comment by AndroidDeveloperLB** [comment]
Have you tried it on the same hardware?
Tested on Pixel 2 with Android P PPR2.180905.005 .

Do you think it's a hardware/OS issue instead?

**Comment by rom1v** [comment]
> Have you tried it on the same hardware?

No.

> Do you think it's a hardware/OS issue instead?

Without any additional info, I would say it's an issue in the Hardware Composer HAL, but I don't know.

**Comment by AndroidDeveloperLB** [comment]
Should I report about this on the issue tracker?
Or maybe try on a different device?

**Comment by mrk-han** [comment]
This also spams me with Scrcpy when using Pixel on 9.0

**Comment by AndroidDeveloperLB** [comment]
@rom1v Why close? Was it fixed?

**Comment by rom1v** [comment]
Because:
> I don't think it's related to scrcpy.
And there is no activity.

**Comment by mrk-han** [comment]
Did you attempt to recreate on a PIxel 2 on 9.0? Looks like both of us had the issue on that exact device

**Comment by mrk-han** [comment]
Maybe you can use your context knowledge of SCRCPY To help us isolate the problem so we can file an issue in the appropriate place? e.g. Google tracker if appropriate

**Comment by rom1v** [comment]
> Did you attempt to recreate on a PIxel 2 on 9.0?

No, I don't have a Pixel 2.

> Looks like both of us had the issue on that exact device

This seems to confirm my intuition that this is a device/ROM issue.

**Comment by AndroidDeveloperLB** [comment]
Actually now that I've tested it, it seems that this log doesn't appear anymore.
There are other stuff, but doesn't seem to be more than what it is when not using scrcpy.

The thing is that this log error appeared very often only when using scrcpy, and I don't think (as I remember) that it even appeared when not using it.

This time I've tested on Pixel 2 with Android Q beta 4. 
I hope it will stay this way on final version. Maybe even better.
